examples/ex15p.cpp

```diff
@@ -316,7 +336,9 @@ int main(int argc, char *argv[])
          refiner.Apply(pmesh);
          if (myid == 0)
          {
-            cout << ", total error: " << estimator.GetTotalError() << endl;
+            //TODO how to handle this?
```

Comment:
This is open as something TODO and related to my first question above in the comments.
`GetTotalError` is neither in the `AbstractErrorEstimator` not in the `ErrorEstimator` class. I would like to move the `GetTotalError` up towards the `AbstractErrorEstimator` or at least into the `ErrorEstimator`, but I am not sure if this is mathematically sound, i.e. if we can get a total error for every estimator. I mean, for the `ErrorEstimator` in the worst case the total error can be bounded by the sum of the local errors. At least if we assume that the error estimator truly bounds the local errors.

I am also a bit confused about the use case for the `AbstractErrorEstimator` class.

Reply:
I think the method `GetTotalError` should be moved to the `ErrorEstimator` class (not sure what the `AbstractErrorEstimator` is for, actually).

Reply:
Going through the ZZ error estimator code again, could it be that the total error is computed wrong? If we have more than 1 process then the total errors are not summed up across the processes, right?

Reply:
The same should be true for LpErrorEstimator.

Reply:
`AbstractErrorEstimator` can serve as a base class for error estimators that compute a different set of errors, not just one double per element.

Moving `GetTotalError` to the `ErrorEstimator` makes sense to me. It's an estimator-specific global measure -- each estimator can use a different way of reducing the element errors into a single global value.

> Going through the ZZ error estimator code again, could it be that the total error is computed wrong?

Indeed, it looks like there is no MPI-global reduction in `ZienkiewiczZhuEstimator` and specifically in the `ZZErrorEstimator` function. We should probably add it. This should not affect the actual refinement results since `GetTotalError` is used just in the `ex15p` to print that number (at least in the `master` branch).

Reply:
@v-dobrev Should I include a patch for this issue in this PR?

Reply:
> @v-dobrev Should I include a patch for this issue in this PR?

Yes, please, if you don't mind.

Reply:
Total error fixed in 8bd7b14 . It does not crash but is not really testable, as the estimator is not working properly in parallel,  yet.